Firestore/Swift/Source/Codable/DocumentSnapshot+ReadDecodable.swift

```diff
@@ -35,11 +35,9 @@ public extension DocumentSnapshot {
   ///     default decoder.
   func data<T: Decodable>(as type: T.Type,
                           with serverTimestampBehavior: ServerTimestampBehavior = .none,
-                          decoder: Firestore.Decoder? = nil) throws -> T? {
+                          decoder: Firestore.Decoder? = nil) throws -> T {
```

Comment:
While we're making the breaking change anyways... think it's worthwhile to make the `decoder` parameter match RTDB and make it non-optional and have the default value be `Firestore.Decoder()`? cc @schmidt-sebastian @peterfriese

Reply:
I like that change. While technically breaking, it will only affect the calling code if it explicitly called it with a `nil` argument.

Reply:
SGTM

Reply:
Implemented with latest commit.